### ISSUE TITLE
I cannot get custom plugins and running workflows at the same time

### ISSUE THREAD

**Issue by list4c** [jeffreytse/jekyll-deploy-action#41]
Hi,
I'm playing with this Github action. It deploys my blog, but the somehow I can't get it to deploy the state of the gh-pages branch.

My repository with the blog is here: https://github.com/list4c/list4c.github.io/tree/gh-pages

In pictures:

![image](https://user-images.githubusercontent.com/100785544/166169158-cd8b3c53-2618-43aa-9ecd-c25b1df6ee7d.png)

This is how the figures look when my publishing source is `main `(set in Repository - > Settings -> Pages)

If I change it to `gh-pages`, then the website content gets updated and the `timeago` plugin works, showing `yesterday` - see here:

![image](https://user-images.githubusercontent.com/100785544/166169139-c6496254-c451-41e8-af45-65af4a5a8822.png)

So I was thinking that I should change it to `gh-pages`, but when I do it, the workflows stop running. So that must be the wrong option. What could I be doing wrong?

So I would like to: have workflows running as defined + have the plugin correctly working on the deployed site, on a continuous basis. I guess something might be set up incorrectly with my deployments.

I'll greatly appreciate your expertise on this matter

**Comment by list4c** [comment]
Hi!
Ok, I found out what was going on :)
I deactivated one of the workflows, thinking that Github Actions somehow 'remembered' another github action that I tried before yours. But in fact the during deployment there are two workflows running - one that builds the site and pushes the output to `gh-pages` branch

![image](https://user-images.githubusercontent.com/100785544/166233263-764331b3-86a9-4703-b579-43751afe0894.png)


 And then the Github-owned workflow which builds and deploys the page from the selected source branch (`gh-pages` here), right?

![image](https://user-images.githubusercontent.com/100785544/166232460-02cf2170-d219-4006-85c0-b370085e0346.png)

So, after activating back that workflow and switching the source to gh-pages branch, everything works like a charm.
